server_src/routes/printers.js

```diff
@@ -125,25 +127,26 @@ router.get("/groups", ensureAuthenticated, async (req, res) => {
 });
 router.post("/printerInfo", ensureAuthenticated, async (req, res) => {
   const id = req.body.i;
+  const returnPrinter = await PrinterClean.returnPrintersInformation(id);
+  res.send(returnPrinter);
+});
 
-  const printers = await PrinterClean.returnPrintersInformation();
-  if (typeof id === "undefined" || id === null) {
-    res.send(printers);
-  } else {
-    const index = _.findIndex(printers, function (o) {
-      //Make sure ID's are both strings to stop recursion issues
-      return o._id == id;
-    });
-    const returnPrinter = {
-      printerName: printers[index].printerName,
-      apikey: printers[index].apikey,
-      _id: printers[index]._id,
-      printerURL: printers[index].printerURL,
-      storage: printers[index].storage,
-      fileList: printers[index].fileList,
-      systemChecks: printers[index].systemChecks,
-    };
-    res.send(returnPrinter);
+router.post("/updatePrinterSettings", ensureAuthenticated, async (req, res) => {
+  const id = req.body.i;
+  if (!id) {
+    logger.error("Printer Settings: No ID key was provided");
+    res.statusMessage = "No ID key was provided";
+    res.sendStatus(400);
+    return;
+  }
+  try {
+    await Runner.getLatestOctoPrintSettingsValues(id);
+    let printerInformation = PrinterClean.returnPrintersInformation(id);
+    res.send(printerInformation);
+  } catch (e) {
+    logger.error(`The server couldn't update your printer settings! ${e}`);
+    res.statusMessage = `The server couldn't update your printer settings! ${e}`;
+    res.sendStatus(500);
   }
 });
```

Comment:
New endpoint to update printer settings, must provide an ID for it. Sends information to client when no id provided with status message. I would like to utilise these status messages a lot more as it will help people using the client in the future to actually get an error response back rather than just a status code.